### PR TITLE
Add and update CDW modules

### PR DESCRIPTION
**Pull request by wmudge**
Dependent on the following PRs:
* https://github.com/cloudera-labs/cdpy/pull/27
* https://github.com/cloudera-labs/cdpy/pull/28

### CODE REVIEW

**Comment by wmudge** [comment]
Either we commit this one or #19 - but not both!

**Comment by wmudge** [comment]
Note that the details returned by `cloudera.cloud.dw_cluster_info` no longer calls `cdpy.gather_clusters` as this call has different values depending on whether or not you pass in `env`. In addition, that call is out-of-scope for the module.

File: plugins/modules/dw_virtual_warehouse.py [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+description:
+    - Create CDP Virtual Warehouse
+author:
+  - "Webster Mudge (@wmudge)"
+  - "Dan Chaffelson (@chaffelson)"
+  - "Saravanan Raju (@raju-saravanan)"
+requirements:
+  - cdpy
+options:
+  id:
```
Comment (Chaffelson) [inline plugins/modules/dw_virtual_warehouse.py:38]: also

File: plugins/modules/dw_virtual_warehouse.py [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+            configBlocks:
+                - id: hive-kerberos-config
+                  format: TEXT
+                  content:
+                      text: "\n[libdefaults]\n\trenew_lifetime = 7d"   
+                      
+# Delete a Virtual Warehouse
+- cloudera.cloud.dw_virtual_warehouse:
+    cluster_id: example-cluster-id
+    id: example-virtual-warehouse-id
```
Comment (Chaffelson) [inline plugins/modules/dw_virtual_warehouse.py:242]: and here

File: plugins/modules/dw_virtual_warehouse.py [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+  elements: str
+'''
+
+
+class DwVw(CdpModule):
+    def __init__(self, module):
+        super(DwVw, self).__init__(module)
+
+        # Set variables
+        self.id = self._get_param('id')
```
Comment (Chaffelson) [inline plugins/modules/dw_virtual_warehouse.py:318]: bare `id`

File: plugins/modules/dw_virtual_warehouse.py [hunk 4]
```diff
@@ -0,0 +1,470 @@
...
+        # Initialize internal values
+        self.target = None
+        self.changed = False
+
+        # Execute logic process
+        self.process()
+
+    @CdpModule._Decorators.process_debug
+    def process(self):
+        if self.id is None:
```
Comment (Chaffelson) [inline plugins/modules/dw_virtual_warehouse.py:348]: bare `id`

File: plugins/modules/dw_virtual_warehouse_info.py [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+description:
+    - Gather information about CDP Virtual Warehouses
+author:
+  - "Webster Mudge (@wmudge)"
+  - "Dan Chaffelson (@chaffelson)"
+  - "Saravanan Raju (@raju-saravanan)"
+requirements:
+  - cdpy
+options:
+  id:
```
Comment (Chaffelson) [inline plugins/modules/dw_virtual_warehouse_info.py:38]: bare `id`

**Review by Chaffelson — CHANGES_REQUESTED** [review]
Mostly the use of bare `id` ambiguously across the modules

File: plugins/modules/dw_database_catalog.py [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+description:
+    - Create, manage, and destroy CDP Data Warehouse Database Catalogs
+author:
+  - "Webster Mudge (@wmudge)"
+  - "Dan Chaffelson (@chaffelson)"
+  - "Saravanan Raju (@raju-saravanan)"
+requirements:
+  - cdpy
+options:
+  id:
```
[the remark this answers is not in the document]
Comment (wmudge) [reply plugins/modules/dw_database_catalog.py:38]: I updated to use `catalog_id` as the name, with `id` as an aliases

File: plugins/modules/dw_database_catalog.py [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+  elements: str
+'''
+
+
+class DwDbc(CdpModule):
+    def __init__(self, module):
+        super(DwDbc, self).__init__(module)
+
+        # Set variables
+        self.id = self._get_param('id')
```
[the remark this answers is not in the document]
Comment (wmudge) [reply plugins/modules/dw_database_catalog.py:135]: Yep, updated this as well

**Pull request merged**